sw/device/lib/testing/i2c_testutils.c

```diff
@@ -164,3 +166,56 @@ bool i2c_testutils_target_check_wr(dif_i2c_t *i2c, uint8_t byte_count,
 
   return i2c_testutils_target_check_end(i2c, cont_byte);
 }
+
+void i2c_testutils_pinmux(const dif_pinmux_t *pinmux, uint8_t kI2cIdx) {
+  top_earlgrey_pinmux_mio_out_t i2c_pinmux_out1_id, i2c_pinmux_out2_id;
+  top_earlgrey_pinmux_insel_t i2c_pinmux_in1_id, i2c_pinmux_in2_id;
+  top_earlgrey_pinmux_peripheral_in_t i2c_pinmux_insel_scl_id,
+      i2c_pinmux_insel_sda_id;
+  top_earlgrey_pinmux_outsel_t i2c_pinmux_outsel_scl_id,
+      i2c_pinmux_outsel_sda_id;
+
+  switch (kI2cIdx) {
+    case 0:
+      i2c_pinmux_out1_id = kTopEarlgreyPinmuxMioOutIoa7;
+      i2c_pinmux_out2_id = kTopEarlgreyPinmuxMioOutIoa8;
+      i2c_pinmux_in1_id = kTopEarlgreyPinmuxInselIoa7;
+      i2c_pinmux_in2_id = kTopEarlgreyPinmuxInselIoa8;
+      i2c_pinmux_insel_scl_id = kTopEarlgreyPinmuxPeripheralInI2c0Scl;
+      i2c_pinmux_insel_sda_id = kTopEarlgreyPinmuxPeripheralInI2c0Sda;
+      i2c_pinmux_outsel_scl_id = kTopEarlgreyPinmuxOutselI2c0Scl;
+      i2c_pinmux_outsel_sda_id = kTopEarlgreyPinmuxOutselI2c0Sda;
+      break;
+    case 1:
+      i2c_pinmux_out1_id = kTopEarlgreyPinmuxMioOutIob9;
+      i2c_pinmux_out2_id = kTopEarlgreyPinmuxMioOutIob10;
+      i2c_pinmux_in1_id = kTopEarlgreyPinmuxInselIob9;
+      i2c_pinmux_in2_id = kTopEarlgreyPinmuxInselIob10;
+      i2c_pinmux_insel_scl_id = kTopEarlgreyPinmuxPeripheralInI2c1Scl;
+      i2c_pinmux_insel_sda_id = kTopEarlgreyPinmuxPeripheralInI2c1Sda;
+      i2c_pinmux_outsel_scl_id = kTopEarlgreyPinmuxOutselI2c1Scl;
+      i2c_pinmux_outsel_sda_id = kTopEarlgreyPinmuxOutselI2c1Sda;
+      break;
+    case 2:
+      i2c_pinmux_out1_id = kTopEarlgreyPinmuxMioOutIob11;
+      i2c_pinmux_out2_id = kTopEarlgreyPinmuxMioOutIob12;
+      i2c_pinmux_in1_id = kTopEarlgreyPinmuxInselIob11;
+      i2c_pinmux_in2_id = kTopEarlgreyPinmuxInselIob12;
+      i2c_pinmux_insel_scl_id = kTopEarlgreyPinmuxPeripheralInI2c2Scl;
+      i2c_pinmux_insel_sda_id = kTopEarlgreyPinmuxPeripheralInI2c2Sda;
+      i2c_pinmux_outsel_scl_id = kTopEarlgreyPinmuxOutselI2c2Scl;
+      i2c_pinmux_outsel_sda_id = kTopEarlgreyPinmuxOutselI2c2Sda;
+      break;
+    default:
+      LOG_FATAL("Unsupported i2c index %d", kI2cIdx);
+  }
```

Comment:
this should probably be an array of structs defined as a static constant in this file.

Reply:
I agree, thanks for the suggestion. It is fixed now.